src/lib/tuning/index/index_evaluator.cpp

```diff
@@ -0,0 +1,76 @@
+#include "index_evaluator.hpp"
+
+#include "optimizer/column_statistics.hpp"
+#include "optimizer/table_statistics.hpp"
+#include "resolve_type.hpp"
+#include "storage/index/base_index.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+#include "types.hpp"
+#include "utils/logging.hpp"
+
+namespace opossum {
+
+IndexEvaluator::IndexEvaluator() {}
+
+void IndexEvaluator::_setup() { _saved_work.clear(); }
+
+void IndexEvaluator::_process_access_record(const BaseIndexEvaluator::AccessRecord& record) {
+  const auto table_statistics = StorageManager::get().get_table(record.column_ref.table_name)->table_statistics();
+  // ToDo(anyone) adapt for multi column indices...
+  const auto predicate_statistics =
+      table_statistics->predicate_statistics(record.column_ref.column_ids[0], record.condition, record.compare_value);
+  const auto total_rows = table_statistics->row_count();
+  const auto match_rows = predicate_statistics->row_count();
+  const auto unscanned_rows = total_rows - match_rows;
+  const float saved_work = unscanned_rows * record.query_frequency;
+  LOG_INFO("saved work for query on " << record.column_ref.table_name << "." << record.column_ref.column_ids[0] << ": "
+                                      << saved_work << "\n");
+  if (_saved_work.count(record.column_ref) > 0) {
+    _saved_work[record.column_ref] += saved_work;
```

Comment:
Having `saved_work` and `_saved_work` which are different things with different types is confusing

Reply:
renamed `_saved_work` to `_saved_work_per_index`